crates/engine/src/memory_engine/transaction.rs

```diff
@@ -0,0 +1,18 @@
+#![allow(clippy::module_name_repetitions)]
+#![allow(clippy::multiple_inherent_impl)]
+
+use crate::TransactionApi;
+
+/// A transaction of the `MemoryEngine`
+#[derive(Copy, Clone, Debug, Default)]
```

Comment:
Why should we derive `Copy` trait and `Clone` trait for `MemorTransaction`? Is the copied or cloned transaction the same as the original transaction?

Reply:
It is temporary because the clippy lint requires it. It will be removed in the next PR.